examples/todoapp/lib/lib_permissions/data_sources/local/permissions_local_data_source.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:realm/realm.dart';
+
+import '../../../base/app/initialization/realm_instance.dart';
+import '../../models/permission_map_model.dart';
+import '../../models/permission_model.dart';
+
+class PermissionsLocalDataSource {
+  PermissionsLocalDataSource({required this.realmInstance});
+  final RealmInstance realmInstance;
+
+  void addPermisions(String id, Map<String, bool> permissions) {
```

Comment:
Considering this method do multiple actions such as adding and updating permissions I would suggest to rename the method name from `addPermisions` to `storePermisions`.

What's the purpose of the `id`  argument ? Do we really need it ? If not feel free to add it as a private constant

Reply:
There must be a primary key property named '_id' on a synchronized Realm

Reply:
I see but, do we need to control it from the upper layers or this should be handled internally by the data source ?

Reply:
Permissions are also part of the flexible sync, so they are saved both locally and on Mongo Atlas, I guess that the id property can be moved to be only within the data source, since it's not used anywhere else in the application